modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/drag_drop/useFileUploader.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {openModal} from 'frontend-js-components-web';
+import React, {useEffect, useState} from 'react';
+
+import {THandleFileDrop} from '../FrontendDataSetDropContext';
+import isFileDropEnabled from '../utils/isFileDropEnabled';
+import {IFileDropSettings} from '../utils/types';
+
+/* This hook connects FDS with state about dropped files, allowing integration
+	with a file uploader component in the future. Current implementation
+	shows a modal with the dropped files.
+ */
+const useFileUploader = ({
+	fileDropSettings,
+	selectedItemsKey,
+}: {
+	fileDropSettings: IFileDropSettings;
+	selectedItemsKey: string | undefined;
+}): {
+	handleFileDrop: (droppedItem: any, dropTarget?: any) => void;
+} => {
+	const [droppedFiles, setDroppedFiles] = useState([]);
+	const [dropTarget, setDropTarget] = useState(null);
+
+	const handleFileDrop: THandleFileDrop = (
+		droppedItem: any,
+		dropTarget?: any
+	) => {
+		if (droppedItem) {
+
+			// @ts-ignore
+
+			const files = droppedItem.files;
+			setDroppedFiles(files);
+			setDropTarget(dropTarget ? dropTarget : null);
+		}
+	};
+
+	useEffect(() => {
+		if (!isFileDropEnabled(fileDropSettings) || !droppedFiles?.length) {
+			return;
+		}
+
+		const ModalBody = () => {
+
+			// @ts-ignore
+
+			const label = (file) =>
+				`'${file.name}' of size '${file.size}' and type '${file.type}'`;
+
+			return (
+				<div>
+					{droppedFiles.map((file) => (
+
+						// @ts-ignore
+
+						<li key={file.name}>{label(file)}</li>
+					))}
+
+					{dropTarget ? (
+						<span>
+							Dropped on item{' '}
+
+							{dropTarget[selectedItemsKey || 'id']}
+						</span>
+					) : (
+						<span>Dropped on the FDS, no specific drop target</span>
+					)}
+				</div>
+			);
+		};
+
+		openModal({
+			bodyComponent: ModalBody,
+			containerProps: {
+				className: 'dsm-actions-icon-selection-modal',
```

Comment:
This class still looks like a copy paste artifact :slightly_smiling_face:

Reply:
fixed